common/unit_test/Test_Padding.cpp

```diff
@@ -43,434 +65,535 @@ void test_reshape1D_1DView(bool is_C2R) {
-  test_reshape1D_2DView(is_C2R);
+TYPED_TEST(GetModifiedShape1D, 2DView) {
+  using float_type = typename TestFixture::float_type;
+  test_reshape1D_2DView<float_type>();
 }
 
-TEST_P(GetModifiedShape1D, 3DView) {
-  bool is_C2R = GetParam();
-  test_reshape1D_3DView(is_C2R);
+TYPED_TEST(GetModifiedShape1D, 3DView) {
+  using float_type = typename TestFixture::float_type;
+  test_reshape1D_3DView<float_type>();
 }
 
-TEST_P(GetModifiedShape1D, 4DView) {
-  bool is_C2R = GetParam();
-  test_reshape1D_4DView(is_C2R);
+TYPED_TEST(GetModifiedShape1D, 4DView) {
+  using float_type = typename TestFixture::float_type;
+  test_reshape1D_4DView<float_type>();
 }
 
-TEST_P(GetModifiedShape1D, 5DView) {
-  bool is_C2R = GetParam();
-  test_reshape1D_5DView(is_C2R);
+TYPED_TEST(GetModifiedShape1D, 5DView) {
+  using float_type = typename TestFixture::float_type;
+  test_reshape1D_5DView<float_type>();
 }
 
-TEST_P(GetModifiedShape1D, 6DView) {
-  bool is_C2R = GetParam();
-  test_reshape1D_6DView(is_C2R);
+TYPED_TEST(GetModifiedShape1D, 6DView) {
+  using float_type = typename TestFixture::float_type;
+  test_reshape1D_6DView<float_type>();
 }
 
-TEST_P(GetModifiedShape1D, 7DView) {
-  bool is_C2R = GetParam();
-  test_reshape1D_7DView(is_C2R);
+TYPED_TEST(GetModifiedShape1D, 7DView) {
+  using float_type = typename TestFixture::float_type;
+  test_reshape1D_7DView<float_type>();
 }
 
-TEST_P(GetModifiedShape1D, 8DView) {
-  bool is_C2R = GetParam();
-  test_reshape1D_8DView(is_C2R);
+TYPED_TEST(GetModifiedShape1D, 8DView) {
+  using float_type = typename TestFixture::float_type;
+  test_reshape1D_8DView<float_type>();
 }
 
-INSTANTIATE_TEST_SUITE_P(CropOrPad, GetModifiedShape1D,
-                         ::testing::Values(true, false));
-
-void test_reshape2D_2DView(bool is_C2R) {
+template <typename T>
+void test_reshape2D_2DView() {
   const int n0 = 30, n1 = 15;
-  View2D<double> x("x", n0, n1);
+  bool is_C2R = !KokkosFFT::Impl::is_complex<T>::value;
+  View2D<T> x_out("x_out", n0, n1);
   constexpr int DIM = 2;
   shape_type<2> default_shape({n0, n1});
 
   for (int axis0 = 0; axis0 < DIM; axis0++) {
     for (int axis1 = 0; axis1 < DIM; axis1++) {
       if (axis0 == axis1) continue;
       axes_type<2> axes({axis0, axis1});
-
+      shape_type<2> in_shape = default_shape;
+      in_shape.at(axis1)     = get_c2r_shape(x_out.extent(axis1), is_C2R);
+      auto [_n0, _n1]        = in_shape;
+      View2D<Kokkos::complex<double>> x_in("x_in", _n0, _n1);
       for (int i0 = -1; i0 <= 1; i0++) {
         for (int i1 = -1; i1 <= 1; i1++) {
           shape_type<2> ref_shape = default_shape;
-          std::size_t n0_new  = static_cast<std::size_t>(x.extent(axis0) + i0);
-          std::size_t n1_new  = static_cast<std::size_t>(x.extent(axis1) + i1);
+          std::size_t n0_new =
+              static_cast<std::size_t>(x_in.extent(axis0) + i0);
+          std::size_t n1_new =
+              static_cast<std::size_t>(x_in.extent(axis1) + i1);
```

Comment:
```suggestion
      auto n0_new = x_in.extent(axis0) + i0;
      auto n1_new = x_in.extent(axis1) + i1;     
```

`static_cast<size_t>` can be done later if needed.